File: java/java-api/src/main/java/com/spring/api/board/service/impl/BoardServiceImpl.java
package com.spring.api.board.service.impl;

import com.spring.api.board.service.BoardService;
import com.spring.jpa.mysql.domain.Board;
import com.spring.jpa.mysql.repository.BoardRepository;
import jakarta.transaction.Transactional;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@Transactional
@AllArgsConstructor
public class BoardServiceImpl implements BoardService {

    private final BoardRepository boardRepository;

    @Override
    public List<Board> getBoardList() {
        this.boardRepository.findAll();
        return List.of();
    }
}

File: kotlin/kotlin-api/src/main/kotlin/com/spring/board/service/impl/BoardServiceImpl.kt
package com.spring.board.service.impl

import com.spring.board.service.BoardService
import com.spring.jpa.mysql.domain.Board
import com.spring.jpa.mysql.repository.BoardRepository
import org.springframework.stereotype.Service

@Service
class BoardServiceImpl(
    private val boardRepository: BoardRepository
) : BoardService {

    override fun getBoardList(): List<Board> {
        this.boardRepository.findAll();
        return listOf()
    }
}